ticdc: add 4 docs for data replication scenarios (#10276)
pingcap/docs-cn: TOC.md

    - [从 CSV 文件迁移数据到 TiDB](/migrate-from-csv-files-to-tidb.md)
    - [从 SQL 文件迁移数据到 TiDB](/migrate-from-sql-files-to-tidb.md)
    - [从 TiDB 集群迁移数据至另一 TiDB 集群](/migrate-from-tidb-to-tidb.md)

>     - [从 TiDB 集群迁移数据至 MySQL 兼容的数据库](/migrate-from-tidb-to-mysql.md)

Reviewer: ```suggestion
    - [从 TiDB 集群迁移数据至兼容 MySQL 的数据库](/migrate-from-tidb-to-mysql.md)
```

    - [下游存在更多列的迁移场景](/migrate-with-more-columns-downstream.md)
    - [如何根据类型或 DDL 内容过滤 binlog 事件](/filter-binlog-event.md)
    - [如何通过 SQL 表达式过滤 DML binlog 事件](/filter-dml-event.md)

> - 数据同步

Reviewer: ```suggestion
- 数据复制
```

>   - [数据同步综述](/replicatation-overview.md)

Reviewer: ```suggestion
  - [数据复制综述](/replication-overview.md)
```

>   - [同步工具](/replication-tools.md)

Reviewer: ```suggestion
  - [复制工具](/replication-tool.md)
```
